tests/common/params_test.py

```diff
@@ -29,6 +29,13 @@ def test_overrides(self):
         assert model_params.pop("type") == "BAR"
         assert model_params["text_field_embedder.tokens.type"] == "BAZ"
 
+    def test_as_flast_dict(self):
+        filename = 'tests/fixtures/bidaf/experiment.json'
+        params = Params.from_file(filename)
```

Comment:
It doesn't look like you're actually testing `as_flat_dict` here.  You're just getting the default `HOCON` behavior.  If you really want to test `as_flat_dict`, also, I'd recommend using a smaller parameter object and testing it exhaustively, instead of just checking for existence of a few keys - the test should also check that there are no _extra_ keys.

Reply:
That's funny--I added a call to `as_flat_dict`.  I agree the test you outlined would be more thorough.